core/src/test/java/org/infinispan/distribution/groups/TransactionalGetGroupKeysTest.java

```diff
@@ -140,8 +140,7 @@ public void testGetGroupsWithConcurrentReplace() throws Exception {
 
       testCache.primaryOwner.put(key(1), value(-1));
 
-      if ((factory == TestCacheFactory.PRIMARY_OWNER || factory == TestCacheFactory.BACKUP_OWNER) &&
-            isolationLevel == IsolationLevel.READ_COMMITTED) {
+      if (isolationLevel == IsolationLevel.READ_COMMITTED) {
```

Comment:
This is specific to GetKeysInGroupCommand: since we don't know the finite set of keys in given group, we have to always send the command to remote nodes to get the whole group. The response will come with the modified value and since we're RC, the value in context gets overwritten.
I had to run the test to realize that, so I'll add a comment there.